app/containers/EditProfilePage/index.js

```diff
@@ -0,0 +1,222 @@
+/**
+ *
+ * EditProfilePage
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import { translationMessages } from 'i18n';
+
+import Profile from 'containers/Profile';
+import * as selectorsProfile from 'containers/Profile/selectors';
+import { makeSelectLocale } from 'containers/LanguageProvider/selectors';
+
+import {
+  DISPLAY_NAME_FIELD,
+  POSITION_FIELD,
+  COMPANY_FIELD,
+  ABOUT_FIELD,
+} from 'containers/Profile/constants';
+
+import {
+  getProfileInfo,
+  getCitiesList,
+  chooseLocation,
+} from 'containers/Profile/actions';
+
+import { getBlob } from 'utils/profileManagement';
+import injectSaga from 'utils/injectSaga';
+import injectReducer from 'utils/injectReducer';
+
+import * as editProfileSelectors from './selectors';
+import reducer from './reducer';
+import saga from './saga';
+
+import {
+  uploadImageFileAction,
+  saveImageChanges,
+  clearImageChanges,
+  setDefaultReducer,
+  saveProfileAction,
+} from './actions';
+
+import ProfileEditForm from './ProfileEditForm';
+
+/* eslint-disable react/prefer-stateless-function */
+export class EditProfilePage extends React.Component {
+  componentWillUpdate(props) {
+    if (props.isOwner === false) {
+      props.history.push('/no-such-page');
```

Comment:
The approach to redirect to non-existing url does not seem clean. Either add route for "Not-found-page" redirect to the page or show some message on the page itself.

Reply:
added route and component talking about lack of rights to complete wrong action